go-controller/pkg/libovsdbops/chassis.go

```diff
@@ -115,3 +140,142 @@ func DeleteChassisWithPredicate(sbClient libovsdbclient.Client, p chassisPredica
+// chassis record in its own local zone Southbound db. ovn-k network controller
+// manager needs to create a chassis records for these remote nodes in its zone Southbound db.
+func CreateOrUpdateRemoteChassis(sbClient libovsdbclient.Client, hostname, name, ip string) error {
+	ch := &sbdb.Chassis{
+		Hostname: hostname,
+		Name:     name,
+	}
+
+	chassis, err := GetChassis(sbClient, ch)
+	if err != nil {
+		if !errors.Is(err, client.ErrNotFound) {
+			return err
+		}
+
+		// Remote chassis record not found. Create it.
+		chassis = &sbdb.Chassis{
+			Hostname: hostname,
+			Name:     name,
+		}
+	}
+
+	if chassis.OtherConfig != nil {
+		if strings.ToLower(chassis.OtherConfig["is-remote"]) == "true" {
+			// Nothing to do. The chassis is already marked as remote.
+			return nil
+		}
+		chassis.OtherConfig["is-remote"] = "true"
+	} else {
+		chassis.OtherConfig = map[string]string{
+			"is-remote": "true",
+		}
+	}
+
+	encap := &sbdb.Encap{
+		Type:        "geneve",
+		ChassisName: name,
+		IP:          ip,
+		Options:     map[string]string{"csum": "true"},
+	}
+	m := newModelClient(sbClient)
+	opModels := []operationModel{
+		{
+			Model:          encap,
+			ModelPredicate: func(item *sbdb.Encap) bool { return item.Type == encap.Type && item.IP == encap.IP },
+			OnModelUpdates: []interface{}{
+				&encap.Options,
+			},
+			DoAfter: func() {
+				chassis.Encaps = []string{encap.UUID}
+			},
+		},
+		{
+			Model:          chassis,
+			ModelPredicate: func(ch *sbdb.Chassis) bool { return ch.Name == name },
+			OnModelUpdates: []interface{}{
+				&chassis.OtherConfig,
+			},
+			OnModelMutations: []interface{}{
+				&chassis.Encaps,
+			},
+		},
+	}
+
+	if _, err := m.CreateOrUpdate(opModels...); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateChassisToLocal marks the chassis record for the node in the Southbound db
+// as a local node.  ie it set "options:is-remote=false" in the node's chassis record.
+// Note that, if "other_config:is-remote" key is not present, it means the chassis
+// is local and just set it explicitly.
+func UpdateChassisToLocal(sbClient libovsdbclient.Client, hostname, name string) error {
+	chassis := &sbdb.Chassis{
+		Hostname: hostname,
+		Name:     name,
+	}
+	chassis, err := GetChassis(sbClient, chassis)
+	if err != nil {
+		return fmt.Errorf("failed to get chassis id %s(%s), error: %v", name, hostname, err)
+	}
+
+	if chassis.OtherConfig != nil {
+		if strings.ToLower(chassis.OtherConfig["is-remote"]) == "false" {
+			// Nothing to do. The chassis is already marked as local.
+			return nil
+		}
+		chassis.OtherConfig["is-remote"] = "false"
+	} else {
+		chassis.OtherConfig = map[string]string{
+			"is-remote": "false",
+		}
+	}
+
+	opModel := operationModel{
+		Model:          chassis,
+		ModelPredicate: func(item *sbdb.Chassis) bool { return item.Name == chassis.Name },
+		OnModelUpdates: onModelUpdatesAllNonDefault(),
```

Comment:
Note that this will not clear the Encaps column if that is something you need to happen.